packages/core/src/util/jwt.ts

```diff
@@ -32,13 +32,22 @@ const logger = createLogger({
  * @returns Decoded payload.
  */
 export function decodeJwt(token: string): DecodedJWT {
-  const decodedToken = jwt.decode(token);
+  return decodeJwtWithHeader(token).payload;
+}
+
+/**
+ * Decode JWT with Header.
+ * @param token - JWT to be decoded
+ * @returns Decoded payload.
+ */
+export function decodeJwtWithHeader(token: string): DecodedJWTWithHeader {
+  const decodedToken = jwt.decode(token, { complete: true });
   if (decodedToken === null || typeof decodedToken === 'string') {
     throw new Error(
       'JwtError: The given jwt payload does not encode valid JSON.'
     );
   }
-  return decodedToken;
+  return { header: decodedToken.header, payload: decodedToken.payload };
```

Comment:
Unfortunately we already used `DecodedJWT`, but wouldn't it make sense to just return the complete decoded JWT?

```suggestion
  return decodedToken;
```

The return type could be `CompleteDecodedJWT`.

Reply:
yes changed it and name the method `decodedJwtComplete` since the natural name was taken.